File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/18/2025, 7:00:55 PM
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        store = {}

        for i, num in enumerate(nums):
            diff = target - num
            if diff in store:
                return [ store[diff], i]
            
            store[num] = i
            

File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/26/2025, 11:37:07 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> store = new HashMap<>();

        for ( int i = 0; i < nums.length; i++ ){
            int difference = target - nums[i];
            if( store.containsKey(difference) ){
                return new int[] { i, store.get(difference) } ;
            };
            store.put( nums[i], i );
        }
        return new int[] {-1, -1};
    }
}